src/bp-xprofile/classes/class-bp-xprofile-group.php

```diff
@@ -352,12 +352,10 @@ public static function get( $args = array() ) {
 			}
 		}
 
-		$signup_fields_order = bp_xprofile_get_signup_field_ids();
-
 		// Pull field objects from the cache.
 		$fields = array();
 		foreach ( $field_ids as $field_id ) {
-			if ( true === $r['signup_fields_only'] && ! in_array( $field_id, $signup_fields_order, true ) ) {
+			if ( true === $r['signup_fields_only'] && ! in_array( $field_id, bp_xprofile_get_signup_field_ids(), true ) ) {
```

Comment:
This was changed here to avoid calling the object cache, or the database, repeatly in the foreach loop.

https://github.com/buddypress/buddypress/commit/15f403f578fed2ee35f3967f50e5510d57395c0a

Reply:
Is this breaking something?

Reply:
Not really, thanks for your comment. As It's generating an error with WP Playground's SQLite db, I was a bit surprised this request was performed even if not on the registration page. I'll try to improve this part differently 👍